src/citrine/resources/dataset.py

```diff
@@ -231,9 +232,14 @@ def register_all(self, data_concepts_resources: List[ResourceType]) -> List[Reso
 
         :return the registered versions
         """
-        return [self.register(resource) for resource
-                in (sorted(data_concepts_resources,
-                           key=lambda resource: writable_sort_order(resource.typ)))]
+        resources = list()
+        for resource in (sorted(data_concepts_resources,
+                         key=lambda resource: writable_sort_order(resource.typ))):
+            registered_resource = self.register(resource)
+            if isinstance(registered_resource, BaseEntity):
+                resource.uids = registered_resource.uids
```

Comment:
Updating uids in place is unexpected behavior that is worth documenting on the method (including a quick explanation of what this supports)

Reply:
Added explanation on behavior to the docstrings